index.js

```diff
@@ -35,16 +35,20 @@ var rethinkdbInit = function (r) {
    */
   var mapIndexes = function (table, db, conn) {
     return table.indexes.map(function (index) {
-     if (typeof index !== 'object' && typeof index !== 'string') throw new TypeError('index entry in table entry must be `Object` or `String`');
-     if (typeof index === 'string') return r.db(db).table(table.name).indexCreate(index).run(conn).catch(existsHandler);
-     if (index.name === undefined) throw new TypeError('index entry object in table schema must have a `name` property');
-     var opts = [];
-     if (index.indexFunction) opts.push(index.indexFunction);
-     if (index.multi || index.geo) opts.push(_.pick(index, ['multi', 'geo']));
-     return r.db(db).table(table.name)
-       .indexCreate(index.name, opts[0], opts[1])
-       .run(conn)
-       .catch(existsHandler);
+      if (typeof index !== 'object' && typeof index !== 'string') throw new TypeError('index entry in table entry must be `Object` or `String`');
+      if (typeof index === 'string') return r.db(db).table(table.name).indexCreate(index).run(conn).catch(existsHandler);
+      if (index.name === undefined) throw new TypeError('index entry object in table schema must have a `name` property');
+      var opts = [];
+      if (index.indexFunction) opts.push(index.indexFunction);
+      if (index.multi || index.geo) opts.push(_.pick(index, ['multi', 'geo']));
+
+      return r.db(db).table(table.name)
+        .indexCreate(index.name, opts[0], opts[1])
+        .run(conn)
+        .catch(existsHandler)
+        .then(function () {
+          return r.db(db).table(table.name).indexWait().run(conn);
```

Comment:
I think this makes sense and it's probably the way it should work. Ran into this myself.

I'm noticing that a lot of these operations are **a lot** slower now than they ware before, so I think these just passed because they weren't a problem (not by design).